### PR TITLE
Explorer optimize startup and testing deadlock

### PR DESCRIPTION
**Pull request by triazo**
Addresses #664 and partially https://github.com/NebulousLabs/Sia-Explorer/issues/11

### CODE REVIEW

File: modules/blockexplorer/addblock.go [hunk 1]
```diff
@@ -172,7 +173,17 @@ func (be *BlockExplorer) addBlockDB(b types.Block) error {
 	}
 	defer tx.Rollback()
 
-	// Construct the struct that will be inside the database
+	// Check if the block is already there before adding it
+	blkBytes, err := be.db.GetFromBucket("Blocks", encoding.Marshal(b.ID()))
+	if err != nil && err != persist.ErrNilEntry {
+		return err
+	}
```
Comment (lukechampine) [inline modules/blockexplorer/addblock.go]: whoops, I didn't realize you already had a `GetFromBucket` function. It kinda overlaps with `getObject`.
fine for now, but maybe later we could add an `exists` helper function that checks for an entry.

File: persist/boltdb.go [hunk 2]
```diff
@@ -26,6 +26,10 @@ type BoltItem struct {
 	Value      []byte
 }
 
+var (
+	ErrNilEntry = errors.New("entry does not exist")
+)
```
Comment (lukechampine) [inline persist/boltdb.go]: I believe an identical error is defined in addblock.go as well, so you can replace those errors with this one.

**Comment by lukechampine** [comment]
build failed; possibly a deadlock timeout?

**Comment by triazo** [comment]
Travis build is now passing

**Pull request merged**